addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/jenkins/JenkinsPipelineRegistry.java

```diff
@@ -13,32 +33,91 @@
  */
 @ApplicationScoped
 public class JenkinsPipelineRegistry {
+    private Map<String, JenkinsPipeline> pipelines = Collections.emptyMap();
+
+    private static final String JENKINS_PIPELINE_REPO_URL = "https://github.com/fabric8io/fabric8-jenkinsfile-library.git";
+    private static final String JENKINS_PIPELINE_REPO_REF = "master";
 
+    private static final Logger log = Logger.getLogger(JenkinsPipelineRegistry.class.getName());
 
     /**
      * Builds the registry index. Should be called only once
      */
     @PostConstruct
     public void index() {
-//        try {
-//            Path pipelinesPath = Paths.get(getClass().getResource("pipelines").toURI());
-//            System.out.println("PIPELINES: " + pipelinesPath);
-//        } catch (URISyntaxException e) {
-//            e.printStackTrace();
-//        }
+        try {
+            Path pipelinesPath = clonePipelinesLibrary(JENKINS_PIPELINE_REPO_URL, JENKINS_PIPELINE_REPO_REF);
+
+            Yaml yaml = new Yaml();
+            Map<String, JenkinsPipeline> pipes =  new HashMap<>();
+            Files.walkFileTree(pipelinesPath, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    String name = file.getFileName().toString();
+                    if (name.equals("metadata.yml") || name.equals("metadata.yaml")) {
+                        JenkinsPipeline p = readPipeline(file);
+                        if (p != null) {
+                            pipes.put(p.getId(), p);
+                        }
+                    }
+                    return super.visitFile(file, attrs);
```

Comment:
No need to call super here, just return `FileVisitResult.CONTINUE`

Reply:
Well that just does the same thing ;)